src/pytest_plugins/forks/forks.py

```diff
@@ -153,6 +170,20 @@ def add_values(self, metafunc: Metafunc, fork_parametrizer: ForkParametrizer) ->
         fork_attribute_name="precompiles",
         parameter_name="precompile",
     ),
+    CovariantDescriptor(
+        marker_name="with_all_evm_code_types",
+        description="marks a test to be parametrized for all EVM code types at parameter named"
+        " evm_code_type of type EVMCodeType",
+        fork_attribute_name="evm_code_types",
+        parameter_name="evm_code_type",
+    ),
+    CovariantDescriptor(
+        marker_name="with_all_call_opcodes",
+        description="marks a test to be parametrized for all *CALL opcodes at parameter named"
+        " call_opcode, and also the appropriate EVM code type at parameter named evm_code_type",
```

Comment:
So call_opcode will be automatically passed as an argument to the test function?
Thats a little implicit. Is there an example how it will look?

Reply:
Yes I added a test as an example to also give myself an idea of how easy would it be to add the marker to any test:
https://github.com/ethereum/execution-spec-tests/blob/a87bb09025d8472c9efacb67ccd839f003211736/tests/cancun/eip4844_blobs/test_point_evaluation_precompile.py#L488-L523

You can see the actual test fixture that consumes the `call_opcode` parameter to prepare the bytecode here:
https://github.com/ethereum/execution-spec-tests/blob/a87bb09025d8472c9efacb67ccd839f003211736/tests/cancun/eip4844_blobs/test_point_evaluation_precompile.py#L144-L173

And my general experience was that it is not so easy and careful thought has to be put when adding this marker.

Main pain points are that:
- The return code is not the same for EOF vs non-EOF call opcodes, but this is somewhat alleviated by this helper function here: https://github.com/ethereum/execution-spec-tests/blob/a87bb09025d8472c9efacb67ccd839f003211736/src/ethereum_test_vm/helpers.py#L8-L20
- The out-of-gas tests simply don't work if we expect to be able to pass an specific amount of gas to a subcall, because the EOF call opcodes don't support a gas parameter.